dex/src/main/scala/com/wavesplatform/dex/api/http/routes/v0/CancelRoute.scala

```diff
@@ -0,0 +1,245 @@
+package com.wavesplatform.dex.api.http.routes.v0
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server._
+import akka.stream.Materializer
+import akka.util.Timeout
+import com.wavesplatform.dex._
+import com.wavesplatform.dex.actors.address.AddressActor
+import com.wavesplatform.dex.api.http.directives.HttpKamonDirectives._
+import com.wavesplatform.dex.api.http.directives.ProtectDirective
+import com.wavesplatform.dex.api.http.entities._
+import com.wavesplatform.dex.api.http.headers.`X-User-Public-Key`
+import com.wavesplatform.dex.api.http.{HasStatusBarrier, _}
+import com.wavesplatform.dex.api.routes.PathMatchers.{AddressPM, AssetPairPM, OrderPM}
+import com.wavesplatform.dex.api.routes.{ApiRoute, AuthRoute}
+import com.wavesplatform.dex.app.MatcherStatus
+import com.wavesplatform.dex.db.OrderDb
+import com.wavesplatform.dex.domain.account.Address
+import com.wavesplatform.dex.domain.asset.AssetPair
+import com.wavesplatform.dex.domain.bytes.ByteStr
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.model.AssetPairBuilder
+import io.swagger.annotations._
+
+import javax.ws.rs.Path
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+@Path("/matcher")
+@Api()
+class CancelRoute(
+  responseTimeout: FiniteDuration,
+  assetPairBuilder: AssetPairBuilder,
+  addressActor: ActorRef,
+  override val matcherStatus: () => MatcherStatus,
+  orderDb: OrderDb[Future],
+  override val apiKeyHash: Option[Array[Byte]]
+)(implicit mat: Materializer)
+    extends ApiRoute
+    with ProtectDirective
+    with HasStatusBarrier
+    with AuthRoute
+    with ScorexLogging {
+
+  implicit private val executionContext: ExecutionContext = mat.executionContext
+  implicit private val timeout: Timeout = responseTimeout
+
+  override lazy val route: Route =
+    pathPrefix("matcher") {
+      pathPrefix("orderbook") {
+        matcherStatusBarrier(cancelOneOrAllInPairOrdersWithSig ~ cancelAllOrdersWithSig)
+      } ~ pathPrefix("orders") {
+        matcherStatusBarrier(cancelOrdersByIdsWithKey ~ cancelOneOrderWithKey)
+      }
+    }
+
+  private def handleCancelRequestToRoute(assetPair: Option[AssetPair], sender: Address, orderId: Option[ByteStr], timestamp: Option[Long]): Route =
```

Comment:
everything in L59:98 can be moved after routes. i think it should slightly improve readability.

Reply:
fixed